### ISSUE TITLE
Provide Architecture Diagram

### ISSUE THREAD

**Issue by toschneck** [kubermatic/kubeone#365]
We should provide some architecture diagram for project newbies to understand how kubeOne creates and manage Clusters. This would also help to explain what's the basic idea behind kubeOne

As blueprint we could use the one cluster API provides: https://github.com/kubernetes-sigs/cluster-api/blob/master/docs/book/common_code/architecture.svg

**Comment by xmudrii** [comment]
Punting to v0.7.0.

/unassign

**Comment by moadqassem** [comment]
![KubeOneArch.png](https://images.zenhubusercontent.com/5d1b48338871121fe293f1ee/f13ab13a-eb29-4668-abc2-de0357bc1d89)

**Comment by cschieder** [comment]
<img width="842" alt="KubeOneDiagram" src="https://user-images.githubusercontent.com/49145827/104206053-659e4780-542f-11eb-95d8-3a652a9dd015.png">

friendly reminder that you have to add the footnote in the wiki. Something like "*As an alternative to the Kubermatic Machine Controller, you can create workers manually and connect them via SSH." maybe link the last part of the sentence to the part of the wiki where this is more closely described

**Comment by xmudrii** [comment]
@cschieder The diagram looks good, however, there are a few things that we can improve IMO:

* Do we really need to showcase providers here? We are not limited to those providers, someone might use KubeOne with any other provider, including using it on bare-metal
* Rename "Master Node" to "Master Instances"
  * The intention is to make it clear that only instances should be created, while KubeOne is going to make a Kubernetes/Master node out of them
* Add a new box between "Master Instances" and "Control Plane" boxes that says "Dependencies/Packages (CRI, kubelet, kubeadm...)"
  * This should be a blue box, i.e. created and managed by KubeOne
* Add a new box between "Control Plane" and "Addons" boxes saying "Features (CNI, metrics-server, NodeLocalDNS...)"
* Add a footnote to "Created by User" that we'll use to mention that KubeOne Terraform Integration and example configs can be used to automate those steps

**Comment by moadqassem** [comment]
> Do we really need to showcase providers here? We are not limited to those providers, someone might use KubeOne with any other provider, including using it on bare-metal.

We added the providers that we officially support. However, I am indifferent to be honest about to leave them or not, maybe it could give a hint to the users, about which cloud providers that we use the best without any surprises so they can follow the same thing.

> Rename "Master Node" to "Master Instances"
I actually got this naming from the `kubeone-config.yaml` file but of course we can change it to master instances.

> Add a new box between "Master Instances" and "Control Plane" boxes that says "Dependencies/Packages (CRI, kubelet, kubeadm...)"

Is this really needed? I would assume being abstract is the key here. I mean a k8s user would assume these things without us telling them that. Plus, there should be some texts underneath the digram which can address these points. IMO diagram should be simple, eventually we are not describing the arch of k8s. 

> Add a new box between "Control Plane" and "Addons" boxes saying "Features (CNI, metrics-server, NodeLocalDNS...)"

Well, if we decide on adding the "Dependencies/Packages" box then I think it makes sense to this as well.

> Add a footnote to "Created by User" that we'll use to mention that KubeOne Terraform Integration and example configs can be used to automate those steps

I agree

## **tl;dr**
I am generally ok with those points that you mentioned and we can add them. Just added those points for any reconsideration ;)

**Comment by cschieder** [comment]
I've been told those are the officially supported providers, if you want we can remove them completely. 
<img width="842" alt="kubeone2" src="https://user-images.githubusercontent.com/49145827/104424562-d2c0f280-557f-11eb-8e24-5d00d7c2564a.png">

**Comment by kron4eg** [comment]
please rename "Master instances" to "Control plane instances" and it's LGTM

**Comment by cschieder** [comment]
<img width="842" alt="kubeone3" src="https://user-images.githubusercontent.com/49145827/105194799-43e14680-5b3a-11eb-8903-1a80f11a7ce9.png">

**Comment by xmudrii** [comment]
https://github.com/kubermatic/docs/pull/502 has got merged
/close

@cschieder Thank you for the diagram! It turned out awesome!
